apis/externalsecrets/v1alpha1/secretstore_kubernetes_types.go

```diff
@@ -0,0 +1,52 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+// Configures a store to sync secrets with a Kubernetes instance.
+type KubernetesProvider struct {
+	// configures the Kubernetes server Address.
+	// +kubebuilder:default= kubernetes.default
+	// +optional
+	Server string `json:"server,omitempty"`
+
```

Comment:
The URL and CA configuration should be on a nested `server` type.

```go
type Server struct {
  
  // configures the Kubernetes server Address.
  // +kubebuilder:default=kubernetes.default
  // +optional
  URL string `json:"url,omitempty"`

  // CABundle is a base64-encoded CA certificate
  // +optional
  CABundle []byte `json:"caBundle,omitempty"`

  // see: https://external-secrets.io/v0.4.1/spec/#external-secrets.io/v1alpha1.CAProvider
  // +optional
  CAProvider CAProvider `json:"caProvider,omitempty"`
 
  // there's still room for impersonation or proxy settings:
  // Impersonate-User
  // Impersonate-Group
  // Impersonate-Extra-( extra name )
  // Impersonate-Uid
  // Proxy Settings
}
```

Reply:
Would the default work as referencing the current Kubernetes cluster?

Reply:
Yes, i think it should by default point to the in-cluster api-server. The CA cert should be taken from a `secretRef` or `serviceAccountRef` - i feel like we shouldn't rely on the `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` mount.

What i would like prevent is that the mounted service account token can be used for authentication because it has access to all the secrets in the cluster (see comment above).

Reply:
The struct already has the `serviceAccountRef`, but right now it only returns a `not implemented` error